### ISSUE TITLE
ods_code and pz_code not set in session if logging in from a given page

### ISSUE THREAD

**Issue by mbarton** [rcpch/national-paediatric-diabetes-audit#180]
To reproduce:

1) Ensure you're logged out
2) Visit `/patients`
3) Log in

Once redirected the switcher should show your PDU but doesn't show anything.

![Screenshot from 2024-07-11 14-45-57](https://github.com/rcpch/national-paediatric-diabetes-audit/assets/395805/f0b8c2ba-3c8e-49ec-9de5-599433d8da17)

**Comment by mbarton** [comment]
It seems like response_url is not login_redirect_url here: https://github.com/rcpch/national-paediatric-diabetes-audit/blob/b227bf35c7aefa732cd0a9c148107f54fa233ee3/project/npda/views/npda_users.py#L519

For example clearing cookies and trying to reload the `/patients` page:

```
!! done: response_url: /patients login_redirect_url: /account/two_factor/
```

I'm having to clear cookies due to the log out button dropdown not appearing (#181)

**Comment by mbarton** [comment]
Fixed in #190
